ui/src/components/retreat/home/PraiseWallContainer.js

```diff
@@ -0,0 +1,282 @@
+          fullName: 'user',
+          message: message,
+          category: eventCategory,
+        });
+        setFormSubmitted(true);
+        GetPraise();
+      } catch (err) {
+        console.log(err);
+      }
+    };
+
+    const [value, setValue] = useState('');
+    const { handleSubmit, register } = useForm();
+
+    return (
+      <Modal isOpen={isOpen} onClose={onClose} size="xl">
+        <ModalOverlay />
+        <ModalContent borderRadius="20">
+          <ModalHeader fontWeight="bold" borderRadius="20">
+            <Icon as={BiNote} w={5} h={5} my="auto" />
+            Submit A Praise!
+            <Text fontSize="sm">
+              Share praises with the church throughout the conference!{' '}
+            </Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <form onSubmit={handleSubmit(PostPraise)}>
+            <ModalBody bgColor="#ebebeb">
+              <Textarea
+                placeholder="Type your praise here..."
+                id="message"
+                size="xl"
+                h="10em"
+                {...register('message')}
+                onChange={setValue}
+              />
+            </ModalBody>
+
+            <ModalFooter bgColor="#ebebeb" borderBottomRadius="20">
+              <Button
+                colorScheme="teal"
+                mx="1em"
+                borderRadius="20"
+                shadow="lg"
+                m="auto"
+                w="10em"
+                type="submit"
+              >
+                <AddIcon w={3} h={3} mx="1" />
+                Submit Praise
+              </Button>
+            </ModalFooter>
+          </form>
+        </ModalContent>
+      </Modal>
+    );
+  };
+  const AlertMessage = () => {
+    const [show, setShow] = useState(true);
+    useEffect(() => {
+      const timeId = setTimeout(() => {
+        setShow(false);
+        setFormSubmitted(false);
+      }, 5000);
+      return () => {
+        clearTimeout(timeId);
+      };
+    }, []);
+    if (!show) {
+      return null;
+    }
+    return (
+      <Alert status="success" borderRadius="20">
+        <AlertIcon />
+        <AlertDescription>Praise submitted successfully!</AlertDescription>
+      </Alert>
+    );
+  };
+  const GetPraise = async () => {
+    try {
+      const { data, status } = await axios.get('/api/praises/get', {
+        params: { category: eventCategory },
+      });
+      if (status === 200) {
+        data.sort((a, b) => (a.createdAt < b.createdAt ? 1 : -1));
+        setPraiseList([...data]);
+      } else {
+        throw Error('Something went wrong');
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const MobileView = () => {
+    return (
+      <Center display={{ base: 'flex', lg: 'none' }}>
+        <HStack align="start" spacing="5">
+          {SplitToChunks(
+            praiseList.filter(
+              (partitionedList) => partitionedList.category == eventCategory
```

Comment:
why have to check if category === eventCategory again? thought already done by API